### PR TITLE
Spark batch ingestion common code abstraction.

### PR DESCRIPTION
**Pull request by abhioncbr**
As per the [issue](https://github.com/apache/pinot/issues/14271), this code abstracts the common code from the Spark 2.4 and Spark 3 modules into common code.

### CODE REVIEW

File: pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-spark-base/pom.xml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  <profiles>
+    <profile>
+      <id>pinot-fastdev</id>
+      <properties>
+        <shade.phase.prop>none</shade.phase.prop>
+      </properties>
+    </profile>
+  </profiles>
+</project>
```
Comment (Jackie-Jiang) [inline pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-spark-base/pom.xml:53]: (minor) Add an empty line

File: pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-spark-2.4/pom.xml [hunk 2]
```diff
@@ -108,6 +108,12 @@
       <groupId>org.codehaus.woodstox</groupId>
       <artifactId>stax2-api</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.apache.pinot</groupId>
+      <artifactId>pinot-batch-ingestion-spark-base</artifactId>
```
Comment (Jackie-Jiang) [inline pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-spark-2.4/pom.xml:113]: You may replace `pinot-batch-ingestion-common` with `pinot-batch-ingestion-spark-base` in the above dependency. No need to specify version or scope explicitly

File: pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-spark-2.4/src/main/java/org/apache/pinot/plugin/ingestion/batch/spark/SparkSegmentMetadataPushJobRunner.java [hunk 3]
```diff
@@ -43,8 +36,8 @@
 import org.apache.spark.api.java.function.VoidFunction;
 
 
-public class SparkSegmentMetadataPushJobRunner implements IngestionJobRunner, Serializable {
-  private SegmentGenerationJobSpec _spec;
+
```
Comment (Jackie-Jiang) [inline pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-spark-2.4/src/main/java/org/apache/pinot/plugin/ingestion/batch/spark/SparkSegmentMetadataPushJobRunner.java:39]: (minor, format) Too many empty lines

File: pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-spark-base/src/main/java/org/apache/pinot/plugin/ingestion/batch/spark/common/AbstractSparkSegmentMetadataPushJobRunner.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+import org.apache.pinot.spi.filesystem.PinotFSFactory;
+import org.apache.pinot.spi.ingestion.batch.runner.IngestionJobRunner;
+import org.apache.pinot.spi.ingestion.batch.spec.Constants;
+import org.apache.pinot.spi.ingestion.batch.spec.PinotFSSpec;
+import org.apache.pinot.spi.ingestion.batch.spec.SegmentGenerationJobSpec;
+import org.apache.pinot.spi.utils.retry.AttemptsExceededException;
+import org.apache.pinot.spi.utils.retry.RetriableOperationException;
+
+
+public abstract class AbstractSparkSegmentMetadataPushJobRunner implements IngestionJobRunner, Serializable {
```
Comment (Jackie-Jiang) [inline pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-spark-base/src/main/java/org/apache/pinot/plugin/ingestion/batch/spark/common/AbstractSparkSegmentMetadataPushJobRunner.java:40]: Suggest calling these abstract classes `BaseSpark...`

**Review by Jackie-Jiang — COMMENTED** [review]
Thanks for the cleanup! Didn't go over the changes in detail. Is the new code identical to the old?

**Comment by Jackie-Jiang** [comment]
@jadami10 Are you using the spark job for the batch ingestion? It would be good if you can help review this PR

**Review by jadami10 — COMMENTED** [review]
this seems right. From what I can tell, it's just code moving.

curious why you're not doing `SparkSegmentGenerationJobRunner.java` and `SparkSegmentMetadataPushJobRunner.java`.

**Comment by abhioncbr** [comment]
> this seems right. From what I can tell, it's just code moving.
> 
> curious why you're not doing `SparkSegmentGenerationJobRunner.java` and `SparkSegmentMetadataPushJobRunner.java`.

Yes, it's just a common code movement; no change in logic. 

It's also there for `SparkSegmentMetadataPushJobRunner` ([reference](pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-spark-base/src/main/java/org/apache/pinot/plugin/ingestion/batch/spark/common/BaseSparkSegmentMetadataPushJobRunner.java)). However, I am not able to find a common code pattern for `SparkSegmentGenerationJobRunner`.

**Comment by abhioncbr** [comment]
Can i get some review on this? Thanks

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**